Error reading JobQueue: "%!s(<nil>)"

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```sh
$ terraform -v
Terraform v0.11.3
```

### Affected Resource(s)

* aws_batch_job_queue

### Terraform Configuration Files

```hcl
resource "aws_batch_job_queue" "job_queue" {
  name     = "${var.name_prefix}-job-queue"
  state    = "ENABLED"
  priority = 1

  compute_environments = [
    "${aws_batch_compute_environment.compute_environment.arn}",
  ]

  depends_on = ["aws_batch_compute_environment.compute_environment"]
}
```

### Debug Output

I can't share the full log due to NDA/secrecy rules at the company I work for. Here are the relevant sections of logs I could find. 

```
2018-10-02T23:01:13.640Z [DEBUG] plugin.terraform-provider-aws_v1.38.0_x4: 2018/10/02 23:01:13 [DEBUG] Job Queue "[redacted]" is already gone
2018/10/02 23:01:13 [ERROR] root.[redacted].batch_compute_environment: eval: *terraform.EvalRefresh, err: aws_batch_job_queue.job_queue: Error reading JobQueue: "%!s(<nil>)"
2018/10/02 23:01:13 [ERROR] root.[redacted].batch_compute_environment: eval: *terraform.EvalSequence, err: aws_batch_job_queue.job_queue: Error reading JobQueue: "%!s(<nil>)"
```

### Expected Behavior

Terraform state should already know this has been removed in a previous execution of `terraform destroy`, or it should not have been removed in a previous execution. subsequent destroys on following failed destroys should not risk an out of date state.

### Actual Behavior

The job queue was destroyed without terraform state being properly updated. In addition to this, the error message should be far more intuitive than "error reading job queue %!s(<nil)".

### Steps to Reproduce

1. `terraform destroy`
2. terraform fails to destroy because there is a running job in the queue therefore the job queue cannot be destroyed.
3. manually terminate the running job via aws console
4. run `terraform destroy` again
5. terraform throws above error.

--- a/aws/resource_aws_batch_job_queue.go
+++ b/aws/resource_aws_batch_job_queue.go
@@ -91,7 +91,8 @@ func resourceAwsBatchJobQueueRead(d *schema.ResourceData, meta interface{}) erro
 		return err
 	}
 	if jq == nil {
-		return fmt.Errorf("Error reading JobQueue: \"%s\"", err)
+		d.SetId("")
+		return nil
 	}
 	d.Set("arn", jq.JobQueueArn)
 	d.Set("compute_environments", jq.ComputeEnvironmentOrder)

